docs/ContinuousIntegration.md

```diff
@@ -0,0 +1,117 @@
+# Firebase Apple SDK Continuous Integration
+
+Firebase uses [several GitHub Action workflows](.github/actions/) for product testing. Most of the
+tests are presubmit tests that are triggered by a PR making changes to any of a
+set of specified files. The rest of the tests, along with the presubmits, are
+scheduled to run once a day. In contrast to the presubmit tests, the nightly
+tests may take a long time to run or be tests that rarely fail.
+
+All presubmits tests being green is a good sign a PR is ready to merge. We do
+allow merges without green tests, because of occasional infrastructure flakiness.
+
+## Organization
+
+Each Firebase product has a corresponding [workflow](firebase-ios-sdk/.github/workflows). There
+are several other workflows primarily for productization and packaging specific testing.
+
+### Product Testing
+
+Each product workflow has several jobs. The jobs typically matrix across the Apple platforms.
+
+#### `pod lib lint` Tests
+
+Use CocoaPods to build, unit test, and run the Xcode Analyzer on the product.
+
+#### SPM Tests
+
+Use Swift Package Manager to build and usually unit test the product.
+
+#### QuickStart Tests
+
+Build and test the product QuickStart.
+
+#### Integration Tests
+
+Run integration tests for the product. These either use a GitHub secret to access a Firebase project
+or use the Firebase emulator
+
+#### Cron Tests
+
+Tests that only run in the nightly cron run. This is typically the `--static-framework` CocoaPods
+configuration that rarely breaks if dynamic frameworks are working.
+
+### Style Tests
+[firebase-ios-sdk/.github/workflows/check.yml](firebase-ios-sdk/.github/workflows/check.yml)
+
+Runs several coding style tests. Details to come in an upcoming document about the scripts
+directory.
+
+### SwiftPM Tests
+[firebase-ios-sdk/.github/workflows/spm.yml](firebase-ios-sdk/.github/workflows/spm.yml)
+
+Build and run Firebase-wide tests with Swift Package Manager.
+
+### Zip Distribution Testing
+[firebase-ios-sdk/.github/workflows/zip.yml](firebase-ios-sdk/.github/workflows/zip.yml)
+
+Builds the zip distribution both from the tip of master and the current staged release distribution.
+The resulting distribution is then used to build and test several Firebase QuickStarts.
+
+### Release testing
+[firebase-ios-sdk/scripts/create_spec_repo](firebase-ios-sdk/scripts/create_spec_repo)
+
+Release testing is to build up a testing podspecs (CocoaPods podspecs) candidate and test building
+up a quickstart with CocoaPods. If this candidate is successfully built, that means tests in
+podspecs, e.g. abtesting test spec, are passed and it is a positive signal to build up a release
+candidate.
+
+Currently we have two workflows running nightly to test podspecs:
+
+#### Release workflow
+[firebase-ios-sdk/.github/workflows/release.yml](firebase-ios-sdk/.github/workflows/release.yml)
+
+The release workflow is to test podspecs on the latest release branch, and create a testing repo.
```

Comment:
w.r.t. "latest release branch"... is that `master` if a release branch does not exist?

Reply:
It should be most recent release tag. I reworded this section. PTAL.